iceoryx_posh/test/moduletests/test_roudi_config_toml_file_provider.cpp

```diff
@@ -14,83 +14,213 @@
 // limitations under the License.
 //
 // SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_dust/cxx/std_string_support.hpp"
 #include "iceoryx_posh/roudi/cmd_line_args.hpp"
 #include "iceoryx_posh/roudi/roudi_config_toml_file_provider.hpp"
 
-#include "iceoryx/tests/posh/moduletests/test_input_path.hpp"
 #include "test.hpp"
 
+#if __cplusplus >= 201703L
+#include <filesystem>
+#endif
+
+#include <fstream>
+#include <string>
+
 namespace
 {
 using namespace ::testing;
 
-using ParseErrorInputFile_t = std::pair<iox::roudi::RouDiConfigFileParseError, iox::roudi::ConfigFilePathString_t>;
+using ParseErrorInputFile_t = std::pair<iox::roudi::RouDiConfigFileParseError, std::string>;
 
 class RoudiConfigTomlFileProvider_test : public TestWithParam<ParseErrorInputFile_t>
 {
   public:
-    void SetUp()
-    {
-        // get file path via cmake
-        m_cmdLineArgs.configFilePath = iox::testing::TEST_INPUT_PATH;
-    }
-
-    void TearDown()
-    {
-    }
-
-    iox::config::CmdLineArgs_t m_cmdLineArgs;
+    iox::config::CmdLineArgs_t cmdLineArgs;
 };
 
 TEST_F(RoudiConfigTomlFileProvider_test, ParseDefaultConfigIsSuccessful)
 {
     ::testing::Test::RecordProperty("TEST_ID", "37a5ad49-1af2-4e4e-a1ba-85826589d560");
     iox::roudi::ConfigFilePathString_t emptyConfigFilePath;
-    m_cmdLineArgs.configFilePath = emptyConfigFilePath;
+    cmdLineArgs.configFilePath = emptyConfigFilePath;
 
-    iox::config::TomlRouDiConfigFileProvider sut(m_cmdLineArgs);
+    iox::config::TomlRouDiConfigFileProvider sut(cmdLineArgs);
 
     auto result = sut.parse();
 
     EXPECT_FALSE(result.has_error());
 }
 
+TEST_F(RoudiConfigTomlFileProvider_test, InvalidPathResultsInError)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "ca2cc3bd-bf39-451c-9dc6-ae90ec0b8ab7");
+    iox::roudi::ConfigFilePathString_t invalidConfigFilePath{"/nowhere/to/find/config.toml"};
+    cmdLineArgs.configFilePath = invalidConfigFilePath;
+
+    iox::config::TomlRouDiConfigFileProvider sut(cmdLineArgs);
+
+    sut.parse()
+        .and_then([](const auto&) {
+            GTEST_FAIL() << "Expected 'RouDiConfigFileParseError::FILE_OPEN_FAILED' but got a config!";
+        })
+        .or_else(
+            [](const auto& error) { EXPECT_THAT(error, Eq(iox::roudi::RouDiConfigFileParseError::FILE_OPEN_FAILED)); });
+}
+
+TEST_F(RoudiConfigTomlFileProvider_test, ParsingFileIsSuccessful)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "37f5a397-a289-4bc1-86a7-d95851a5ab47");
+
+#if __cplusplus < 201703L
+    GTEST_SKIP() << "The test uses std::filesystem which is only available with C++17";
+#else
+    auto tempFilePath = std::filesystem::temp_directory_path();
+    tempFilePath.append("test_roudi_config.toml");
+
+    std::fstream tempFile{tempFilePath, std::ios_base::trunc | std::ios_base::out};
+    ASSERT_TRUE(tempFile.is_open());
+    tempFile << R"([general]
+version = 1
+
+[[segment]]
+
+[[segment.mempool]]
+size = 128
+count = 1
+)";
```

Comment:
Indentation issue?

Reply:
Yes, that would be great. Just wondering: Is there a way to include this in the `.clang-format` file?

Reply:
Done. Even if it would be possible to add it to `.clang-format`, it would defeat the purpose of the raw string literal, meaning to not modify the string in any way. In the context of the TOML parser it does not matter but it might be in other cases